src/liberty/devCommands.ts

```diff
@@ -490,28 +490,35 @@ export async function openReport(reportType: string, libProject?: LibertyProject
         if (path !== undefined) {
             let report: any;
             if (libProject.getContextValue() === LIBERTY_MAVEN_PROJECT || libProject.getContextValue() === LIBERTY_MAVEN_PROJECT_CONTAINER) {
-                report = Path.join(path, "target", "site", reportType + "-report.html");
+                report = getReportFile(path,"reports",reportType+".html");
             } else if (libProject.getContextValue() === LIBERTY_GRADLE_PROJECT || libProject.getContextValue() === LIBERTY_GRADLE_PROJECT_CONTAINER) {
                 report = await getGradleTestReport(libProject.path, path);
             }
             let reportTypeLabel = reportType;
             if (reportType === "gradle") {
                 reportTypeLabel = "test";
             }
-            fs.exists(report, (exists) => {
-                if (exists) {
-                    const panel = vscode.window.createWebviewPanel(
-                        reportType, // Identifies the type of the webview. Used internally
-                        libProject.getLabel() + " " + reportTypeLabel + " report", // Title of the panel displayed to the user
-                        vscode.ViewColumn.Two, // Open the panel in the second window
-                        {}, // Webview options
-                    );
-                    panel.webview.html = getReport(report); // display HTML content
-                } else {
-                    const message = localize("test.report.does.not.exist.run.test.first", report);
-                    vscode.window.showInformationMessage(message);
+            /*
+            if its a maven project, check for the report in the initial path , else try for the alternate one and if it doesnt exist in both 
+            display a prompt for its non existance
+            */
+            if(libProject.getContextValue() === LIBERTY_MAVEN_PROJECT || libProject.getContextValue() === LIBERTY_MAVEN_PROJECT_CONTAINER){
+                console.log("report path ::"+report)
+                if(!await checkReportAndDisplay(report,reportType,reportTypeLabel,libProject)){
+                    report = getReportFile(path,"site",reportType+"-report.html");
+                    if(!await checkReportAndDisplay(report,reportType,reportTypeLabel,libProject)){
+                        const message = localize("test.report.does.not.exist.run.test.first", report);
+                        vscode.window.showInformationMessage(message);
+                    }     
                 }
-            });
+            /*
+            if its a gralde project, then check existance for the report in the path set and if not display prompt for its non existance
+            */
+            }else if(!await checkReportAndDisplay(report,reportType,reportTypeLabel,libProject)){
```

Comment:
Checking for the report and displaying a "no test report found" message is currently implemented in both Maven and Gradle. Could we move this duplicate code into "checkReportAndDisplay"? Additionally, we could add a condition so that if the report type is "Maven" and the path is "reports/", the message is not displayed.

Reply:
Made the changes as suggested , please [review](https://github.com/JoshwinThomasIBM/liberty-tools-vscode/commit/4f54c54fd8367ecfa5a5a8ef21fed3d061ed306c)